eqs/src/query_result_state.rs

```diff
@@ -44,6 +51,8 @@ pub struct QueryResultState {
     // additional indexed data for queries
     pub transaction_by_id: HashMap<(u64, u64), CommittedCapeTransition>,
     pub transaction_id_by_hash: HashMap<Commitment<CapeTransition>, (u64, u64)>,
+
+    pub system_status: SystemStatus,
```

Comment:
I wonder if this should actually be set back to Initializing on Deserialize...

Reply:
Should be possible with `#[serde(skip)]` and `#[serde(default = "assign_initializing")]` with an `fn assign_initializing() -> SystemStatus { SystemStatus::Initializing }`

Reply:
Oh, good catch! Loading system_status of Available from persistent store doesn't make sense if there might be an unbounded amount of work to do after deserialization.